zephyr/esp32s3/CMakeLists.txt

```diff
@@ -119,32 +119,32 @@ if(CONFIG_SOC_SERIES_ESP32S3)
   zephyr_library_sources_ifdef(CONFIG_COUNTER_TMR_ESP32 ../../components/hal/timer_hal.c)
 
   zephyr_sources_ifdef(
-    CONFIG_I2C_ESP32
+    CONFIG_USE_ESP32_HAL_I2C
     ../../components/hal/i2c_hal_iram.c
     ../../components/hal/i2c_hal.c
     )
 
   zephyr_library_sources_ifdef(CONFIG_WDT_ESP32 ../../components/hal/wdt_hal_iram.c)
 
   zephyr_sources_ifdef(
-    CONFIG_PWM_LED_ESP32
+    CONFIG_USE_ESP32_HAL_LED_PWM
     ../../components/soc/esp32s3/ledc_periph.c
     ../../components/hal/ledc_hal_iram.c
     ../../components/hal/ledc_hal.c
     )
 
   zephyr_sources_ifdef(
-    CONFIG_MCPWM_ESP32
+    CONFIG_USE_ESP32_HAL_MCPWM
     ../../components/hal/mcpwm_hal.c
     )
 
   zephyr_sources_ifdef(
-    CONFIG_PCNT_ESP32
+    CONFIG_USE_ESP32_HAL_PCNT
     ../../components/hal/pcnt_hal.c
     )
 
   zephyr_sources_ifdef(
-    CONFIG_ADC_ESP32
+    CONFIG_USE_ESP32_HAL_ADC
     ../../components/hal/adc_hal.c
     ../esp_shared/components/driver/adc_common.c
     ../esp_shared/components/driver/rtc_io_hal.c
```

Comment:
ADC implies rtc stuff? looks wrong

Reply:
`rtc_io` here is the subsystem responsible for handling low power and gpio configurations attach to RTC power domain. So it is indeed required.

Reply:
so it should be ADC selecting it, no?